overlord/state/change.go

```diff
@@ -630,3 +631,75 @@ func (c *Change) abortTasks(tasks []*Task, abortedLanes map[int]bool, seenTasks
 		c.abortLanes(lanes, abortedLanes, seenTasks)
 	}
 }
+
+// VerifyTaskDependencies checks the tasks in the change for cyclic dependencies
+// and returns in error in such case.
+func (c *Change) VerifyTaskDependencies() error {
+	tasks := c.Tasks()
+	// list of successors of a task (their IDs)
+	successors := make(map[string][]string, len(tasks))
+	// count incoming edges to a task (how many tasks precede a given task)
```

Comment:
I don't find this comment very clear: "precede" has a temporal meaning, and "incoming edges" on the other hand suggest that we are looking for direct dependencies only.

Suggestion: make it clear whether we are talking of direct dependencies only, or if indirect dependencies count too. Then maybe "require" is better than "precede" (but there's no need to rename the variables: "predecessors" and "successors" are fine, once their meaning is well defined by the comments).

Reply:
how many tasks the task waits for ?